File: lib/models/transaction.dart
class Transaction {
  final String id;
  final String title;
  final double amount;
  final DateTime date;
  final bool isCredit;

  Transaction({
    required this.id,
    required this.title,
    required this.amount,
    required this.date,
    required this.isCredit,
  });
}

File: backend/models/Transaction.js
const mongoose = require('mongoose');

const TransactionSchema = new mongoose.Schema({
    sender: { type: String, required: true },   // Wallet Address / Email
    receiver: { type: String, required: true }, // Wallet Address / Email
    amount: { type: Number, required: true },
    timestamp: { type: Date, default: Date.now },
    status: { type: String, default: 'SUCCESS' },
    hash: { type: String, required: false }, // Simulated Block Hash
    txnId: { type: String, required: true, unique: true } // Unique Transaction ID
});

module.exports = mongoose.model('Transaction', TransactionSchema);
